src/pages/[platform]/build-a-backend/more-features/analytics/analytics-migration-guide/index.mdx

```diff
@@ -26,110 +26,492 @@ export function getStaticProps(context) {
-
-```ts
-import { record, flushEvents } from 'aws-amplify/analytics/kinesis';
-
-record({
-  data: {
-    // The data blob to put into the record
-  },
-  partitionKey: 'myPartitionKey',
-  streamName: 'myKinesisStream'
-});
-
-flushEvents();
-```
-Finally, let's look at the difference in how to personalize recommendations,
-
-```diff
-- import { Analytics, AmazonPersonalizeProvider } from 'aws-amplify';
-- Analytics.addPluggable(new AmazonPersonalizeProvider());
-
-+ import { record } from 'aws-amplify/analytics/personalize';
-
-
-- Analytics.record({
--     eventType: "Identify",
--     properties: {
--         "userId": "<USER_ID>"
--     }
-- }, 'AmazonPersonalize');
-
-+ record({
-+   eventType: "Identify",
-+   properties: {
-+      userId: "<USER_ID>"
-+   }
-+ });
-
-```
-Notice that in v5, the `Analytics.record` API takes in a second parameter with the string value `AmazonPersonalize` whereas in v6, the `record` function is imported from the `aws-amplify/analytics/personalize` subpath and takes only one parameter. 
+This guide will help you migrate Amplify JavaScript v5's Analytics APIs to the new v6's Analytics APIs.
+
+As of v6 of Amplify, you will now import APIs directly from the `aws-amplify/analytics` path when using the Pinpoint provider. If you are using the Personalize, Kinesis, or Kinesis (Stream) provider, you will need to import separate APIs from the appropriate subpath (ie `import { record } from 'aws-amplify/analytics/personalize'` when using the `record` API with the `AmazonPersonalize` provider) instead of passing in the `provider` or `options.provider` parameter.
+
+<Callout>TODO: The `startSession` API has been removed in v6. (Due to auto-track capabilities?)</Callout>
+
+## Enable and Disable
+
+In both v5 and v6, Analytics is enabled by default. You can disable and enable all Analytics providers using the `enable` and `disable` APIs exported from the `aws-amplify/analytics` path.
+
+<BlockSwitcher>
+  <Block name="V6">
+    ```ts
+    import { enable, disable } from 'aws-amplify/analytics';
+
+    // disable all Analytics providers (enabled by default)
+    disable();
+
+    // enable all Analytics providers
+    enable();
+    ```
+
+  </Block>
+  <Block name="V5">    
+    ```ts
+    import { Analytics } from 'aws-amplify';
+
+    Analytics.disable();
+
+    Analytics.enable();
+    ```
+
+  </Block>
+</BlockSwitcher>
+
+## Pinpoint
+
+In v6, `provider` is determined by import path. To use Pinpoint APIs you will import them directly from `aws-amplify/analytics`. These APIs no longer accept parameters specific to Kinesis or Personalize.
+
+### Analytics.record
+
+The `record` API in v6 has not changed in behavior, however, there are a couple of changes to be aware of in v6:
+
+- The `record` API is now synchronous and does not return as events are always buffered.
```

Comment:
Does not return what?

Reply:
Promise<unknown>, I'll try to clarify in the verbiage.